src/Symfony/Component/Serializer/Normalizer/ObjectNormalizer.php

```diff
@@ -162,4 +129,62 @@ public function denormalize($data, $class, $format = null, array $context = arra
 
         return $object;
     }
+
+    /**
+     * Gets and caches attributes for this class and context.
+     *
+     * @param object $object
+     * @param array  $context
+     *
+     * @return array
+     */
+    private function getAttributes($object, array $context)
+    {
+        $key = sprintf('%s-%s', get_class($object), serialize($context));
+
+        if (isset(self::$attributesCache[$key])) {
+            return self::$attributesCache[$key];
+        }
+
+        if (self::$attributesCache[$key] = $this->getAllowedAttributes($object, $context, true)) {
```

Comment:
you should not assign false here. It could corrupt the cache in case an exception is thrown in the following logic, is catched, and the method is called again: as you assign false in the cache array, the cache is set and used on future calls (and breaks the contracts of the method). Use a local variable and assign the cache only on next line when using the value.

and the logic is also altered here: an empty array should be returned here. Only ``false`` triggers the fallback logic. So you need to use a strict comparison